plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/jdt/ui/wizards/NewInterfaceCreationWizardPage.java

```diff
@@ -0,0 +1,253 @@
+
+/**
+ * Wizard page for creating a java interface.
+ * 
+ * @author zcervink@redhat.com
+ * 
+ */
+public class NewInterfaceCreationWizardPage extends WizardPage {
+
+	/**
+	 * Instantiates a new new java interface wizard page.
+	 */
+	public NewInterfaceCreationWizardPage(ReferencedComposite referencedComposite) {
+		super(referencedComposite);
+	}
+
+	/**
+	 * Sets a given name.
+	 * 
+	 * @param name Name
+	 */
+	public NewInterfaceCreationWizardPage setName(String name) {
+		new LabeledText(this, "Name:").setText(name);
+		return this;
+	}
+
+	/**
+	 * Sets a given package name.
+	 * 
+	 * @param packageName Package name
+	 */
+	public NewInterfaceCreationWizardPage setPackage(String packageName) {
+		new LabeledText(this, "Package:").setText(packageName);
+		return this;
+	}
+
+	/**
+	 * Sets a given source folder.
+	 * 
+	 * @param sourceFolder Source folder
+	 */
+	public NewInterfaceCreationWizardPage setSourceFolder(String sourceFolder) {
+		new LabeledText(this, "Source folder:").setText(sourceFolder);
+		return this;
+	}
+
+	/**
+	 * Returns a package name.
+	 * 
+	 * @return package name
+	 */
+	public String getPackage() {
+		return new LabeledText(this, "Package:").getText();
+	}
+
+	/**
+	 * Returns a interface name.
+	 * 
+	 * @return interface name
+	 */
+	public String getName() {
+		return new LabeledText(this, "Name:").getText();
+	}
+
+	/**
+	 * Returns a source folder.
+	 * 
+	 * @return source folder
+	 */
+	public String getSourceFolder() {
+		return new LabeledText(this, "Source folder:").getText();
+	}
+
+	/**
+	 * Check/Uncheck Enclosing type.
+	 *
+	 * @param toggle enclosing type checkbox
+	 */
+	public NewInterfaceCreationWizardPage toggleEnclosingTypeCheckbox(boolean toggle) {
+		new CheckBox(this, "Enclosing type:").toggle(toggle);
+		return this;
+	}
+
+	/**
+	 * Returns Enclosing type.
+	 * 
+	 * @return enclosing type
+	 */
+	public String getEnclosingType() {
+		return new DefaultText(2).getText();
+	}
+
+	/**
+	 * Sets a given Enclosing type.
+	 * 
+	 * @param enclosingTypeName enclosing type
+	 */
+	public NewInterfaceCreationWizardPage setEnclosingType(String enclosingTypeName) {
+		new DefaultText(2).setText(enclosingTypeName);
+		return this;
+	}
+
+	/**
+	 * Gets an enabled modifier.
+	 * 
+	 * @param String modifier name
+	 */
+	public String getCurrentModifier() {
+		if (new RadioButton(this, "public").isSelected()) {
+			return "public";
+		} else if (new RadioButton(this, "package").isSelected()) {
+			return "package";
+		} else if (new RadioButton(this, "private").isSelected()) {
+			return "private";
+		} else if (new RadioButton(this, "protected").isSelected()) {
+			return "protected";
+		}
+
+		throw new RedDeerException(
+				"New Java Interface wizard modifiers property has to be set to 1 of 4 values ('public', 'package', 'private' or 'protected').");
+	}
+
+	/**
+	 * Sets a given modifier.
+	 * 
+	 */
+	public NewInterfaceCreationWizardPage setPublicModifier() {
```

Comment:
Can you please update javadoc comment for all methods for setting modifier? e.g. from Sets a given modifier. to Sets modifier to public.

Reply:
Updated.